orbstation/spells/eldritch_minion.dm

```diff
@@ -0,0 +1,252 @@
+	summoned.alpha = 0
+	summoned.notransform = TRUE
+	summoned.move_resist = MOVE_FORCE_OVERPOWERING
+	animate(summoned, 10 SECONDS, alpha = 155)
+
+	message_admins("A [summoned.name] is being summoned by [ADMIN_LOOKUPFLW(user)] in [ADMIN_COORDJMP(summoned)].")
+	var/list/mob/dead/observer/candidates = poll_candidates_for_mob("Do you want to play as a [summoned.real_name]?", ROLE_HERETIC, FALSE, 10 SECONDS, summoned)
+	if(!LAZYLEN(candidates))
+		summon_location.balloon_alert(user, "ritual failed, no ghosts!")
+		animate(summoned, 0.5 SECONDS, alpha = 0)
+		QDEL_IN(summoned, 0.6 SECONDS)
+		return FALSE
+
+	var/mob/dead/observer/picked_candidate = pick(candidates)
+	// Ok let's make them an interactable mob now, since we got a ghost
+	summoned.alpha = 255
+	summoned.notransform = FALSE
+	summoned.move_resist = initial(summoned.move_resist)
+
+	summoned.ghostize(FALSE)
+	summoned.key = picked_candidate.key
+
+	user.log_message("created a [summoned.name], controlled by [key_name(picked_candidate)].", LOG_GAME)
+	message_admins("[ADMIN_LOOKUPFLW(user)] created a [summoned.name], [ADMIN_LOOKUPFLW(summoned)].")
+
+	var/datum/antagonist/heretic_monster/heretic_monster = summoned.mind.add_antag_datum(/datum/antagonist/heretic_monster)
+	heretic_monster.set_owner(user.mind)
+
+	return TRUE
+
+// Not sure why you'd pick this but I think it should be on the list anyway
+/datum/demonological_summon/shade
+	name = "Shade"
+	desc = "A corporeal spirit which moves very quickly and can travel gaseously through vents. A fragile scout. \
+		Its fickle loyalty can be borrowed by anyone bearing a Soul Shard."
+	icon = 'icons/mob/nonhuman-player/cult.dmi'
+	icon_state = "shade_cult"
+	mob_path = /mob/living/simple_animal/shade
+
+// I'm not including juggernaut or artificer frankly because they're just a bit boring
+/datum/demonological_summon/wraith
+	name = "Wraith"
+	desc = "A shell of mystic metal animated by a bound spirit. It can move quickly and travel through walls. \
+		A fragile assassin."
+	icon = 'icons/mob/nonhuman-player/cult.dmi'
+	icon_state = "wraith"
+	mob_path = /mob/living/simple_animal/hostile/construct/wraith/mystic
+
+/// I suspect the most popular one
+/datum/demonological_summon/prophet
+	name = "Raw Prophet"
+	desc = "A constructed of stitched-together limbs which tumbles end-over-end. It can see and travel through walls and \
+		link people together in a psychic web. A fragile scout."
+	icon_state = "raw_prophet"
+	mob_path = /mob/living/simple_animal/hostile/heretic_summon/raw_prophet
+
+/datum/demonological_summon/rust_spirit
+	name = "Rust Spirit"
+	desc = "A mechanical beast animated by a bound spirit. It spreads rust to destroy structures and machines, and can expel rust at its enemies \
+		to poison and disorient them. Heals while in rusted areas. A brawler, and not a creature of subtlety."
+	icon_state = "raw_prophet"
+	mob_path = /mob/living/simple_animal/hostile/heretic_summon/raw_prophet
```

Comment:
You've forgotten to set the proper mob paths for rust and ash spirits - they'll only summon Raw Prophets. The icon for rust spirits is also wrong.

Reply:
I am going to refactor this so that errors will be more immediately obvious.

Reply:
Never mind, I cannot use `initial()` to populate these properties without actually instantiating an instance of the type 😔

Reply:
Never mind the never mind, I figured it out by finding a BYOND forum post from 2012 which seems to be the only place documenting the required implementation.